test/functional/page_objects/discover_page.ts

```diff
@@ -152,178 +137,163 @@ export function DiscoverPageProvider({ getService, getPageObjects }) {
-      log.debug('theHeight = ' + yAxisHeight);
-      const bars = $('g > g.series > rect')
-        .toArray()
-        .map(chart => {
-          const barHeight = $(chart).attr('height');
-          return Math.round((barHeight / yAxisHeight) * yAxisLabel);
-        });
-
-      return bars;
-    }
-
-    async getChartInterval() {
+    public async getChartInterval() {
       const selectedValue = await testSubjects.getAttribute('discoverIntervalSelect', 'value');
-      const selectedOption = await find.byCssSelector('option[value="' + selectedValue + '"]');
+      const selectedOption = await find.byCssSelector(`option[value="${selectedValue}"]`);
       return selectedOption.getVisibleText();
     }
 
-    async setChartInterval(interval) {
-      const optionElement = await find.byCssSelector('option[label="' + interval + '"]', 5000);
+    public async setChartInterval(interval: string) {
+      const optionElement = await find.byCssSelector(`option[label="${interval}"]`, 5000);
       await optionElement.click();
-      return await PageObjects.header.waitUntilLoadingHasFinished();
+      return await header.waitUntilLoadingHasFinished();
     }
 
-    async getHitCount() {
-      await PageObjects.header.waitUntilLoadingHasFinished();
+    public async getHitCount() {
+      await header.waitUntilLoadingHasFinished();
       return await testSubjects.getVisibleText('discoverQueryHits');
     }
 
-    async getDocHeader() {
-      const header = await find.byCssSelector('thead > tr:nth-child(1)');
-      return await header.getVisibleText();
+    public async getDocHeader() {
+      const docHeader = await find.byCssSelector('thead > tr:nth-child(1)');
+      return await docHeader.getVisibleText();
     }
 
-    async getDocTableIndex(index) {
-      const row = await find.byCssSelector('tr.kbnDocTable__row:nth-child(' + index + ')');
+    public async getDocTableIndex(index: number) {
+      const row = await find.byCssSelector(`tr.kbnDocTable__row:nth-child(${index})`);
       return await row.getVisibleText();
     }
 
-    async getDocTableField(index) {
+    public async getDocTableField(index: number) {
       const field = await find.byCssSelector(
         `tr.kbnDocTable__row:nth-child(${index}) > [data-test-subj='docTableField']`
       );
       return await field.getVisibleText();
     }
 
-    async clickDocSortDown() {
+    public async clickDocSortDown() {
       await find.clickByCssSelector('.fa-sort-down');
     }
 
-    async clickDocSortUp() {
+    public async clickDocSortUp() {
       await find.clickByCssSelector('.fa-sort-up');
     }
 
-    async getMarks() {
-      const marks = await find.allByCssSelector('mark');
-      return await Promise.all(marks.map(mark => mark.getVisibleText()));
+    public async getMarks() {
+      const table = await docTable.getTable();
+      const $ = await table.parseDomContent();
+      return $('mark')
+        .toArray()
+        .map(mark => $(mark).text());
```

Comment:
What's the reason for the switch to `parseDomContent`  change? code becomes more verbose? thx!

Reply:
To speedup and minimize flackiness:
```
      const marks = await find.allByCssSelector('mark');
      return await Promise.all(marks.map(mark => mark.getVisibleText()));
```

With the code above `_field_data` test webdriver makes 51 rest call (1 - /elements, 50 for each element) The main problem is that while iterating through array of 50 elements DOM might be updated and `mark.getVisibleText()` will fail since element reference is no longer valid.

With the new code we make 2 rest calls (1 - /element, 1 - /innerHTML), faster and since we search in DOM snapshot there is no longer dependency UI rendering.

Reply:
it is not the best approach for every case, but here `mark` is in the span and I don't see a reason why taking text from DOM can be different from text we are getting with webdriver rest call.

Reply:
thank you very much for the explanation!